library/psa_crypto_slot_management.c

```diff
@@ -278,11 +278,10 @@ static psa_status_t persistent_key_setup( psa_key_lifetime_t lifetime,
 #endif /* !defined(MBEDTLS_PSA_CRYPTO_STORAGE_C) */
 }
 
-psa_status_t psa_open_key( psa_key_lifetime_t lifetime,
-                           psa_key_file_id_t id,
-                           psa_key_handle_t *handle )
+psa_status_t psa_open_key( psa_key_file_id_t id, psa_key_handle_t *handle )
 {
-    return( persistent_key_setup( lifetime, id, handle, PSA_SUCCESS ) );
+    return( persistent_key_setup( PSA_KEY_LIFETIME_PERSISTENT,
+                                  id, handle, PSA_SUCCESS ) );
```

Comment:
Would nobody ever open a key that doesn't have a `#PSA_KEY_LIFETIME_PERSISTENT` lifetime? Is open not used for volatile keys? Is this obvious enough from the function documentation?

Would anybody ever try this, (perhaps from separate threads or processes)?
1. psa_create_key(PSA_KEY_LIFETIME_VOLATILE, someid, &handle1);
1. psa_open_key(someid, &handle2);

Two issues:
- They might not realize that the create does an implicit open (although this is documented already)
- Key ownership and sharing implications potentially come into scope, in a service model. e.g. if we allow opening of a volatile key, do we need to block opening a volatile key that is already open? do we share the key somehow?

Reply:
Volatile keys don't have a key id. I think that's now clearly explained in the doc of `psa_key_id_t`. Maybe it needs to be explained elsewhere as well?

Since volatile keys don't have a key id, you can't open them.

Reply:
What's the connection with sharing? Why would sharing require a persistent id?

What's the current API for key sharing ? We discussed proposals in https://github.com/ARMmbed/psa-crypto/issues/72 , but that was done by the services team, and I can't find any design document now.

Reply:
How can you know you are opening the same volatile key without having a persistent ID? Do we prevent opening multiple handles to the same key?

Reply:
You can't open a volatile key. Opening requires a persistent key id.

Reply:
Yes, understood. Because you can't open a volatile key, I'm highlighting potential difficulty in obtaining multiple handles to a volatile key. It's easy to get multiple handles to a persistent key; I don't believe we prevent obtaining multiple handles to a persistent key.

> Why would sharing require a persistent id?
Because you can't open a volatile key.

Reply:
There's no way to obtain multiple handles to the same volatile key. Sharing creates a separate handle which (at least conceptually) points to a copy of the key material.

Reply:
Thanks for answering my questions. Sharing is out of scope for this PR, so resolving this conversation.